### PR TITLE
feature; Dynamic container loading support & Error collection

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
- `storage` package config accepts an init container. It only allows reading and writing data from the init container. This PR expands the capability of optional loading arbitrary containers during the runtime. This is controlled using a bool flag, which is default disabled, to maintain backwards compatibility.

- We also add error collection to a common errors package

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/297
&
https://github.com/lyft/flyte/issues/321

## Follow-up issue
NA

### CODE REVIEW

File: storage/stowstore.go [hunk 1]
```diff
@@ -36,6 +37,10 @@ var fQNFn = map[string]func(string) DataReference{
 	azure.Kind: func(bucket string) DataReference {
 		return DataReference(fmt.Sprintf("afs://%s", bucket))
 	},
+	local.Kind: func(bucket string) DataReference {
+		// Eventually we may want to make this file://
```
Comment (katrogan) [inline storage/stowstore.go:41]: I don't understand this comment, is that not the case below?

Comment (kumare3) [reply storage/stowstore.go:41]: aah older comment, should delete it actually

Comment (kumare3) [inline storage/stowstore.go:41]:
```suggestion
```

Comment (kumare3) [reply storage/stowstore.go]: done

File: config/errros.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package config
```
Comment (wild-endeavor) [inline config/errros.go:1]: filename has a typo in it.

File: config/errros.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+import "fmt"
+
+var (
+	ErrStrictModeValidation       = fmt.Errorf("failed strict mode check")
```
Comment (wild-endeavor) [inline config/errros.go:6]: I know this was in the original file https://github.com/lyft/flytestdlib/blob/master/config/errors.go#L6 but is it intentional that we don't want to use the `ErrorCode` class that flytestdlib defines?

File: config/errros.go (hunk 2 above)
Comment (kumare3) [reply config/errros.go:1]: ohh that was always bad ha, let me fix it

File: config/errros.go (hunk 3 above)
Comment (kumare3) [reply config/errros.go:6]: no we want the error specifically to compare i think

File: storage/localstore_test.go [hunk 4]
```diff
@@ -7,22 +7,25 @@ import (
 	"path/filepath"
```
Comment (wild-endeavor) [inline storage/localstore_test.go:7]: if `localstore.go` was deleted, can we rename this file?

File: config/errros.go (hunk 2 above)
Comment (kumare3) [reply config/errros.go:1]: done

File: storage/localstore_test.go (hunk 4 above)
Comment (kumare3) [reply storage/localstore_test.go:7]: no this is fine, it is still testing localstore

File: storage/config.go [hunk 5]
```diff
@@ -39,15 +39,17 @@ var (
...
 type Config struct {
-	Type          Type             `json:"type" pflag:",Sets the type of storage to configure [s3/minio/local/mem/stow]."`
-	Connection    ConnectionConfig `json:"connection"`
-	Stow          *StowConfig      `json:"stow,omitempty"`
-	InitContainer string           `json:"container" pflag:",Initial container to create -if it doesn't exist-.'"`
+	Type                  Type             `json:"type" pflag:",Sets the type of storage to configure [s3/minio/local/mem/stow]."`
+	Connection            ConnectionConfig `json:"connection"`
+	Stow                  *StowConfig      `json:"stow,omitempty"`
+	InitContainer         string           `json:"container" pflag:",Initial container to create -if it doesn't exist-.'"`
+	MultiContainerEnabled bool             `json:"enable-multicontainer" pflag:",If this is true, then the container argument is overlooked and redundant. This config will automatically open new connections to new containers/buckets as they are encountered"`
```
[the remark this answers is not in the document]
Comment (kumare3) [reply storage/config.go:52]: If you look at the previous field, it is called Container. Sad that is the nomenclature. In Stow, container is a bucket and we decided to go with that. I can see the confusion

File: storage/stowstore.go [hunk 6]
```diff
@@ -6,6 +6,7 @@ import (
```
[the remark this answers is not in the document]
Comment (kumare3) [reply storage/stowstore.go:6]: haha, ya I think I can maybe collapse them
